cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -3199,4 +3199,23 @@ Result<std::shared_ptr<io::OutputStream>> AzureFileSystem::OpenAppendStream(
   return impl_->OpenAppendStream(location, metadata, false, this);
 }
 
+Result<std::string> AzureFileSystem::PathFromUri(const std::string& uri_string) const {
+  std::string path;
+  Uri uri;
+  RETURN_NOT_OK(uri.Parse(uri_string));
+  RETURN_NOT_OK(AzureOptions::FromUri(uri, &path));
+
+  std::vector<std::string> supported_schemes = {"abfs", "abfss"};
```

Comment:
nit: It would be nice to have a comment explaining why `internal::PathFromUriHelper()` can't be used in this case.

Reply:
Good idea, I added a comment. :+1: